lib/patches/tenant_runner.rb

```diff
@@ -8,19 +9,22 @@ def initialize(path: nil, tenants: nil)
 
   def perform
     Patches.logger.info("Patches tenant runner for: #{tenants.join(',')}")
-
     tenants.each do |tenant|
-      Apartment::Tenant.switch(tenant)
-      runner = build
-      runner.perform
+      if parallel?
+        Patches::TenantWorker.perform_async(tenant, path)
+      else
+        run(tenant, path)
```

Comment:
Just my 2c, I'd rather avoid using Concerns (a Ruby module in this case) just for the sake of making classes smaller. This is purely a cosmetic change and adds a cognitive overhead finding methods scattered across multiple files.

Reply:
It's not being used for the sake of making the class smaller but re-using the method.

Reply:
Hmm why does it need to be reused? Wouldn't this work?
```ruby
if parallel?
  Patches::TenantWorker.perform_async(tenant, path)
else
  Patches::TenantWorker.new.perform(tenant, path)
end
```

Reply:
We can't use the worker for both cases cause Sidekiq is not a mandatory dependency. You should be able to use patches without it. At least in theory given the check for defined?(Sidekiq) that I found at some point.